docs/monitoring/tag-support.md

```diff
@@ -5,47 +5,64 @@ description: Asssign metadata to host/server/container with custom tags and crea
+SPM supports 2 types of tags:
 
-SPM client has the ability to:
+1. Physical tags -  Physical Tag is an attribute of a data point one can use for filtering and grouping. They are sent with every data point. They can either automatically collected by agent or can be configured. e.g hostname, jvm name etc.
+2. Logical tags - Logical tags are stored just once and updated periodically and are not sent as part of every data point. They are associated with a set of physical tags. Once can filter/group data points using logical tags without sending them with every data point. e.g. cloud tags
 
-1.  Collect AWS/EC2 tags and send them to SPM. EC2 Instances should be
-    created with AWS IAM Role that has policy 'AmazonEC2ReadOnlyAccess'.
-     See [AWS/EC2 User Guide](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html)
-2.  Send EC2 Instance Type and Availability Zone as tags.
+## **Cloud Tags**
 
-### **Excluding Specific Tags**
+SPM client has the ability to collect metadata as tags from AWS, Azure and GCE instances. Cloud tags are sent as logical tags. 
+SPM collects below metadata:
 
-To exclude tags and thus not send them to SPM just edit the monitor
-configuration file -
-/opt/spm/spm-monitor/conf/spm-monitor-config-${token}-${jvm}.properties:
+1. InstanceId
+2. InstanceName (Azure and GCE)
+3. InstanceType
+4. Region (AWS & Azure)
+5. AvailabilityZone (AWS & GCE)
+6. ProjectID (GCE)
+5. User defined tags
+
+To collect user defined tags you need below IAM roles:
+
+1. AWS - EC2 Instances should be created with AWS IAM Role that has policy `AmazonEC2ReadOnlyAccess`.
+    See [AWS/EC2 User Guide](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) for more info.
+2. Azure - To fetch resource tags for Virtual Machines, you need to grant `Reader` role to its Resource Group in Azure Resource Manager
+    See [Access Azure Resource Manager API](https://docs.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/tutorial-linux-vm-access-arm) for more info.
+3. GCE - In GCE User defined tags are called as Labels. To read labels, the instance needs `roles/compute.viewer` IAM role.
+    See [Granting Roles to Service Accounts](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource) for more info.
+
+Cloud tag collection is enabled by default.  To disable Cloud tags
+collection adjust set `cloud.metadata-enabled` to `false` in `/opt/spm/properties/sta.yml`.
+
+### **Custom Tags**
+
+Sematext-agent-java support configuration of custom physical tags. To add custom tags for each app edit the below
+property in the monitor configuration file: /opt/spm/spm-monitor/conf/spm-monitor-config-${token}-${jvm}.properties:
 
 ``` properties
-# uncomment and add tags which should be excluded
-# SPM_SUPPRESS_TAGS=project:baz, node:qux
+# add tags if you want to use them, example: SPM_MONITOR_TAGS="env:foo, role:bar"
```

Comment:
Here in the actual config and maybe in this doc, should we say what the valid tag format is, what valid characters are, what characters a tag can start with, if spaces or special characters are allowed, if one can have a tag value with a : character, if there is a max length for key or value, if tags are case sensitive, etc.?

Reply:
I think they are not, right @prog8 ?
But I don't know if we lowercase them all, of example.

Reply:
we don't apply any changes on tags so they are case sensitive in a way that if one send "TaG" and then "tag" we'll keep them separately and present in UI separately.

Reply:
Btw. why don't we lowercase them all?  Wouldn't that help people avoid inconsistency in their tags?